lib/ansible/modules/windows/win_psmodule.ps1

```diff
@@ -0,0 +1,181 @@
+#!powershell
+# This file is part of Ansible
+#
+# Copyright 2017, Daniele Lazzari <lazzari@mailup.com>
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+# win_psmodule (Powershell modules Additions/Removal)
+
+$params = Parse-Args $args -supports_check_mode $true
+
+$name = Get-AnsibleParam -obj $params "name" -type "str" -failifempty $true
+$repo = Get-AnsibleParam -obj $params "repository" -type "str" 
+$url = Get-AnsibleParam -obj $params "url" -type "str"
+$state = Get-AnsibleParam -obj $params "state" -type "str" -default "present" -validateset "present", "absent"
+$allow_clobber = Get-AnsibleParam -obj $params "allow_clobber" -type "bool" -default $false
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -default $false
+
+$result = @{"changed" = $false
+            "output" = ""}
+
+
+Function Install-NugetProvider {
+  param(
+    [bool]$CheckMode
+    )
+  $PackageProvider = Get-PackageProvider -ListAvailable|?{($_.name -eq 'Nuget') -and ($_.version -ge "2.8.5.201")}
+  if (!($PackageProvider)){
+      try{
+        Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force -ErrorAction Stop -WhatIf:$CheckMode | out-null
```

Comment:
Yes, if it changes something on the system, it should result in a change. win_chocolatey does something similar. One could argue if a module should do anything more than managing the object(s) it has control over, but for convenience sometimes more is managed.